sympy/series/formal.py

```diff
@@ -0,0 +1,445 @@
+    def start(self):
+        return self.interval.inf
+
+    @property
+    def stop(self):
+        return self.interval.sup
+
+    @property
+    def length(self):
+        return oo
+
+    @property
+    def infinite(self):
+        """returns an infinite representation of the series"""
+        from sympy.concrete import Sum
+
+        ak, xk = self.ak, self.xk
+        k = ak.variables[0]
+
+        return self.ind + Sum(ak.formula * xk.formula, (k, ak.start, ak.stop))
+
+    def polynomial(self, n=6):
+        """truncated series as polynomial.
+
+        returns series sexpansion of f upto order ``O(x**n)``
+        as a polynomial (without ``O`` term)
+        """
+        terms = []
+        for i, t in enumerate(self):
+            if i >= n:
+                break
+            if t is not S.Zero:
+                terms.append(t)
+
+        return Add(*terms)
+
+    def truncate(self, n=6):
+        """truncated series.
+
+        returns truncated series expansion of f upto
+        order ``O(x**n)``
+
+        if n is ``None``, returns an infinite iterator
+        """
+        if n is None:
+            return iter(self)
+
+        x, x0 = self.x, self.x0
+        pt_xk = self.xk.coeff(n)
+        if x0 is S.NegativeInfinity:
+            x0 = S.Infinity
+
+        return self.polynomial(n) + Order(pt_xk, (x, x0))
+
+    def _eval_term(self, pt):
+        pt_xk = self.xk.coeff(pt)
+
+        def _get_xterm(t):
+            if self.x0 in [S.Infinity, -S.Infinity]:
+                t = t.as_numer_denom()[1]
+            else:
+                t = t.as_numer_denom()[0]
+            return t.as_independent(self.x)[1]
+
+        ind = S.Zero
+        if self.ind:
+            for t in Add.make_args(self.ind):
+                xterm = _get_xterm(t)
+                if pt_xk == 1 and xterm == 1:
+                    ind += t
+                elif xterm != 1 and pt_xk != 1:
+                    j = xterm.as_base_exp()[1]
+                    if j == pt:
+                        ind += t
+
+        try:
+            pt_ak = self.ak.coeff(pt).simplify() # TODO: Thoughts?
+        except IndexError:
+            pt_ak = S.Zero
+
+        term = (pt_ak * pt_xk) + ind
+
+        return term.collect(self.x)
+
+    def _eval_subs(self, old, new):
+        x = self.x
+        if old.has(x):
+            return self
+
+    def _eval_as_leading_term(self, x):
+        for t in self:
+            if t is not S.Zero:
+                return t
+
+
+def fps(f, x=None, x0=0, dir=1, hyper=True, order=4, rational=True, full=False):
+    """Generates Formal Power Series of f
+
+    returns the formal series expansion of ``f`` around ``x = x0``
+    with respect to ``x`` in the form of a ``FormalPowerSeries`` object
+
+    Formal Power Series is represented using an explicit formula
+    computed using different algorithms.
+
+    See :func:`compute_fps` for the more details regarding the computation
+    of formula.
+
+    Arguments
+    =========
```

Comment:
SymPy tries to follow the numpydoc standard for docstrings. Please read this: https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt